### DISCUSSION TITLE
Operator precedence between + and ?

### DISCUSSION THREAD

**Discussion by obones** [merbanan/rtl_433#2360] [General]
Hello,

Using Visual Studio to compile rtl_433, I get the following warning for this [line](https://github.com/merbanan/rtl_433/blob/master/src/devices/honeywell_cm921.c#L310):

    Warning	C6336	Arithmetic operator has precedence over question operator, use parentheses to clarify intent.	

The code is this:

      msg->unparsed_length = (num_unparsed_bits / 8) + (num_unparsed_bits % 8) ? 1 : 0;

Right now, this means the addition is done and if it's not zero then `unparsed_length` will be `1` else `0`.
This looks weird as the original intent, I sense that the point was to get the number of bytes required to store `num_unparsed_bits` bits.
If my feeling is right, then the operation can be written like this: 

    msg->unparsed_length = (num_unparsed_bits + 7) >> 3;

In my `manchester_bitrow` branch, I even have a macro to do it:

    // Round up (NUM_BITS / 8)
    #define NUM_BYTES(NUM_BITS) (((NUM_BITS) + 7) >> 3)

Let me know if you feel that a change should be done, I can create the PR.

**Comment by zuckschwerdt** [comment]
Yes canonical code for this is `(x + 7) / 8` or `(x + 7) >> 8` (one or the other might produce better code or confuse the compiler, but only if x is signed).
Please change those instances with prejudice.

**Reply by obones** [reply]
Done: https://github.com/merbanan/rtl_433/pull/2378
